aikau/src/main/resources/alfresco/forms/controls/NumberSpinner.js

```diff
@@ -109,7 +120,7 @@ define(["alfresco/forms/controls/BaseFormControl",
             // See AKU-341 for details of why we handle commas and spaces...
             var value = this.wrappedWidget.textbox.value.replace(",","").replace(" ","");
             var parsedValue = parseFloat(value);
-            var isValid = !isNaN(parsedValue);
+            var isValid = !isNaN(parsedValue) || (lang.trim(value) === "" && this.permitEmpty);
```

Comment:
Very minor: would this be more performant if the options were reversed in the second part of the OR: (this.permitEmpty && lang.trim(value) === "")